ra/1.1/ra-metadata-example.xml

```diff
@@ -1,103 +1,191 @@
+     was deprecated (in multiple translations).
+-->
+<parameter name="foo">
+  <deprecated>
+    <replaced-with name="mode"/>
+    <desc lang="en">Don't use foo, it's bad.</desc>
+    <desc lang="cs">Nepoužívej foo, sic to schytáš</desc>
+  </deprecated>
 <longdesc lang="en">
-The mount options used for mounting a filesystem; normally this is set to
-defaults, but you may want to modify this if you require a read-only
-mount or something similar.
+Whether the example daemon should operate with foo factor
 </longdesc>
-<shortdesc lang="en">Mount options for this filesystem</shortdesc>
-<content type="string" default="defaults" />
+<shortdesc lang="en">Foo factor</shortdesc>
+<content type="string" />
 </parameter>
 
 </parameters>
 
-<!-- List the actions supported by the RA -->
+<!-- All actions supported by the RA must be listed in meta-data. Timeouts are
+     given in seconds (or s, m, h, d suffix with the usual meanings) and should
+     be a reasonable hint how long a certain action might take in the worst
+     case.
+-->
 <actions>
-<!-- Valid actions: start, stop, recover, status, reload, verify-all
-     
-     The timeout is given in seconds (or s, m, h, d postfix and their
-     usual meanings) and should be a reasonable _hint_ to the RM how
-     long a certain action might take in the worst case. 
-  -->
-<action name="start"   timeout="2m30s" />
+
+<!-- The OCF RA API mandates support for start, stop, monitor, and meta-data
+-->
+<action name="start"   timeout="2m" />
 <action name="stop"    timeout="100" />
+<action name="meta-data"  timeout="5" />
+
+<!-- Monitor actions should specify an interval, which is a hint for how often
+     the service status should be polled. Monitor may optionally support
+     multiple depths and/or roles, in which case a separate entry with a unique
+     interval should be given for each.
+-->
+<action name="monitor" depth="0"  timeout="20" interval="10" />
+<action name="monitor" depth="10" timeout="60" interval="1h" role="promoted" start-delay="1m" />
+<action name="monitor" depth="20" timeout="2m" interval="1d" start-delay="2m" />
+
+<!-- The OCF RA API specifies several actions that may optionally be supported
+     but should have a specific meaning if so.
+-->
 <action name="recover" timeout="150" />
-<!-- In what intervals the RM should poll the RA for status; and how early
-     after the start of the RA is should start. Again, this are just
-     _defaults_ and can be overridden by the RM. -->
-<action name="status" depth="0"  timeout="20" interval="10" start-delay="1m" />
-<action name="status" depth="10" timeout="60" interval="1h" start-delay="5m" />
-<action name="status" depth="20" timeout="2m" interval="1d" start-delay="1d" />
 <action name="reload"  timeout="60" />
-<action name="meta-data"  timeout="5" />
-<!-- As with all not mandatory actions, this one is only listed if supported
-  -->
-<action name="verify-all"  timeout="30" />
+<action name="reload-params" timeout="10" />
```

Comment:
It is confusing, and I want the standard to be clear, so thanks for bringing this up.

The ocf:heartbeat:apache agent currently doesn't support the reload action or have any reloadable parameters, but I'll use it as an example of how it could:

The "reload" action should reload the service itself. The apache agent could implement this as "apachectl graceful". The user would call the reload action (directly or via a higher-level tool) after editing the httpd configuration files.

The "reload-params" action should make changes in reloadable parameters effective. An example would be the apache agent's "client" parameter, which can be set to wget or curl, specifying which command the agent uses to check the server's status page. The server doesn't need to be restarted or reloaded for the agent to use a different client, in fact the server doesn't need to know about the change at all. In this example, the apache agent doesn't need to do anything special for a change in client, so the reload-params action would just return success. Pacemaker would call the reload-params action if client changed, rather than a stop and start, which is what it does for parameters not marked reloadable.

How would you suggest wording that?

Reply:
> How would you suggest wording that?

So if I understand correctly, it's actually an action that reloads parameters related to the resource agent itself. While reading your example, it may seem like this is a noop action. But I suppose this might be useful to some agent if they need to eg. create/update links, files, local state, etc.

Maybe the action itself is not well named? What about `reload-agent`?

But then, as the `unique` attribute is deprecated, it's still not clear to me when the RM decide to call `reload` and when it decides to call `reload-params` after a RA configuration update. What about the new `reloadable` being a list with values `agent` and `resource`?

Reply:
> So if I understand correctly, it's actually an action that reloads parameters related to the resource agent itself. While reading your example, it may seem like this is a noop action. But I suppose this might be useful to some agent if they need to eg. create/update links, files, local state, etc.

Exactly
 
> Maybe the action itself is not well named? What about `reload-agent`?

That's a good idea. I don't think anyone has done anything with reload-params yet, so unless anyone has objections, I'll make that change.
 
> But then, as the `unique` attribute is deprecated, it's still not clear to me when the RM decide to call `reload` and when it decides to call `reload-params` after a RA configuration update. What about the new `reloadable` being a list with values `agent` and `resource`?

The RM will never need to call reload, only reload-agent (whenever a reloadable parameter changes). It's expected that Pacemaker's crm_resource will implement something like a --force-reload option comparable to --force-start, --force-stop, etc., tp let the user manually reload any resource, and higher-level tools like pcs, crm shell, GUIs, etc., might offer the capability as well.

Reply:
> The RM will never need to call reload

I suppose it could if some of the agent parameters are related to or embed resource setups for some reasons. I don't want to overthink or bikeshed, as I have no example in mind, but it feels plausible.

Reply:
Changed with new "Rename reload-params action" commit